### ISSUE TITLE
Unrecognized R6 class

### ISSUE THREAD

**Issue by pkrog** [r-lib/lintr#2546]
```r
R/request_fcts.R|177 col 10 warning| [object_usage_linter] no visible binding for global variable 'RequestResult'                                                                              
||   return(RequestResult$new(content = res$content, retry = res$retry,
||          ^~~~~~~~~~~~~
```

The R6 class `RequestResult` resides in a separate file, `RequestResult.R`.
I've tried to insert a roxygen2 line before the internal function containing the culprit line, but it seems `lintr` does not take it into account:
```
#' @include RequestResult.R
```

**Comment by MichaelChirico** [comment]
could you (1) link the package in question and (2) describe how you're running lint()?

lint() doesn't load packages, you need to run it in a session where the package is installed or already loaded (e.g. with pkgload::load_all())

**Comment by MichaelChirico** [comment]
No, that's just something {codetools} does:

```r
f=function(x) Request
g=function(x) pkg:::Request
codetools::checkUsage(f)
# <anonymous>: no visible binding for global variable ‘Request’
codetools::checkUsage(g)
# (nothing)
```

The package needs to be loaded before running `lint_package()`, so this should work:

```r
pkgload::load_all()
lintr::lint_package()
```

**Comment by pkrog** [comment]
I see, thanks.
